File: app/backend/java/ride-service/src/main/java/com/srbms/rideservice/entity/OtpVerification.java
package com.srbms.rideservice.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "otp_verification")
public class OtpVerification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "ride_id")
    private Integer rideId;

    @Column(name = "phone")
    private String phone;

    @Column(name = "otp_code", nullable = false)
    private String otpCode;

    @Column(name = "status")
    private String status; // PENDING, VERIFIED, EXPIRED

    @Column(name = "created_at", insertable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "expires_at")
    private LocalDateTime expiresAt;

    public OtpVerification() {}

    public OtpVerification(Integer rideId, String phone, String otpCode, String status, LocalDateTime expiresAt) {
        this.rideId = rideId;
        this.phone = phone;
        this.otpCode = otpCode;
        this.status = status;
        this.expiresAt = expiresAt;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getRideId() {
        return rideId;
    }

    public void setRideId(Integer rideId) {
        this.rideId = rideId;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getOtpCode() {
        return otpCode;
    }

    public void setOtpCode(String otpCode) {
        this.otpCode = otpCode;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getExpiresAt() {
        return expiresAt;
    }

    public void setExpiresAt(LocalDateTime expiresAt) {
        this.expiresAt = expiresAt;
    }
}

File: app/backend/.net/SmartRide.RideService/Models/Entities/OtpVerification.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartRide.RideService.Models.Entities;

[Table("otp_verification")]
public class OtpVerification
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("id")]
    public int Id { get; set; }

    [Column("ride_id")]
    public int? RideId { get; set; }

    [Column("phone_number")]
    public string? PhoneNumber { get; set; }

    [Column("otp_code")]
    public string OtpCode { get; set; } = string.Empty;

    [Column("status")]
    public string Status { get; set; } = "PENDING";

    [Column("expires_at")]
    public DateTime? ExpiresAt { get; set; }

    public OtpVerification() { }

    public OtpVerification(int? rideId, string? phoneNumber, string otpCode, string status, DateTime? expiresAt)
    {
        RideId = rideId;
        PhoneNumber = phoneNumber;
        OtpCode = otpCode;
        Status = status;
        ExpiresAt = expiresAt;
    }
}
